CheatSheets/html-cheatsheet.md

```diff
@@ -150,11 +153,88 @@
 
-| Command            | Description                                                                      |
-| ------------------ | -------------------------------------------------------------------------------- |
-| `<form> … </form>` | The parent tag for an HTML form.                                                 |
-| `action=”url”`     | The URL listed here is where the form data will be submitted once user fills it. |
-| `method=””`        | It specifies which HTTP method (POST or GET) would be used to submit the form.   |
+```html
+<form method="POST" action="/page">
+  <label for="name">Page Name</label>
+  <input id="name" type="text" name="page_name" />
+  <input type="submit" value="Create" />
+</form>
+```
+
+#### `<form>`'s Attributes
+
+- **method**: the HTTP verb (method) that the browser uses to submit the form. We use "POST" to stand in for `POST`, `PUT` and `DELETE`.
+- **action**: the path of the HTTP request page that processes the information submitted via the form.
+- **the _route_**: is the combination of the verb (method) and path, and must be defined on our app for this form to submit data correctly.
+
+**[🔼Back to Top](#table-of-contents)**
+
+
+### The `<label>` Element (Tag)
+
+> The `<label>` element is the formal way to define a label for an HTML form widget. 
+> This is the most important element if you want to build accessible forms." *— MDN*
+
+There are two ways to use labels correctly:
+
+```html
+<!-- Simple (nested) label example -->
+<label>Click me 
+  <input type="text" id="user" name="name" />
+</label>
+
+<!-- Using the "for" attribute with the input's id -->
+<label for="user">Click me</label>
+<input id="user" type="text" name="name" />
+```
+
+#### `<label>`'s Attributes
+
+**Very important!** `for=` in a label references an `<input>`s `id=` attribute, not it's `name=` attribute!
+Sometimes these values will be the same, but `for=` goes with `id=`. `name=` is the key of the `<input>`'s 
+value when it arrives at the server.
+
+**[🔼Back to Top](#table-of-contents)**
+
+
+## INPUT Types
+
+| Field Type | HTML Code  Notes |
+|:-- |:-- |:--  |
+| plain text | `<input type="text">` | the type attribute can be omitted |
+| password field | `<input type="password">`  | echoes dots instead of characters |
+| text area | `<textarea></textarea>` |  a more customizable plain text area |
+| checkbox | `<input type="checkbox">` | can be toggled on or off |
+| radio button | `<input type="radio">`  can be grouped with other inputs |
+| drop-down lists | `<select><option>` | [check here for more info](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select) |
+| file picker | `<input type="file">`  | pops up an “open file” dialog |
+| hidden field | `<input type="hidden">`   | nothing there!
+| submit button | `<input type="submit">`  | activates the form's submission <br/>(a `POST` request or <br/>Javascript action) |
```

Comment:
Table format is broken [here](https://github.com/ritwikdalmia/CheatSheets-for-Developers/blob/ritwikdalmia-html-cheetsheet/CheatSheets/html-cheatsheet.md#input-types)